GeneratorInterface/GenFilters/interface/PythiaDauFilter.h

```diff
@@ -39,30 +40,27 @@ namespace edm {
   class HepMCProduct;
 }
 
-class PythiaDauFilter : public edm::EDFilter {
+class PythiaDauFilter : public edm::global::EDFilter<> {
    public:
       explicit PythiaDauFilter(const edm::ParameterSet&);
-      ~PythiaDauFilter();
+      ~PythiaDauFilter() override;
 
 
-      virtual bool filter(edm::Event&, const edm::EventSetup&);
+      bool filter(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
    private:
       // ----------memeber function----------------------
 
       // ----------member data ---------------------------
       
-       edm::EDGetTokenT<edm::HepMCProduct> token_;
+       const edm::EDGetTokenT<edm::HepMCProduct> token_;
        std::vector<int> dauIDs;
-       int particleID;
-       bool chargeconju; 
-       int ndaughters;
-       double minptcut;
-       double maxptcut;
-       double minetacut;
-       double maxetacut;
+       const int particleID;
+       const bool chargeconju; 
+       const  int ndaughters;
+       const double minptcut;
+       const double maxptcut;
+       const double minetacut;
+       const double maxetacut;
+       std::auto_ptr<Pythia8::Pythia> fLookupGen; // this instance is for accessing particleData information
```

Comment:
Better to be `std::unique_ptr` since the C++ standard is doing away with `std::auto_ptr`.